### PR TITLE
sql: Update common interfaces

### PR DESCRIPTION
**Pull request by nkprince007**
Let's just stick to the stuff already provided by `database/sql` standard package.

### CODE REVIEW

File: sql/sql.go [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+// DB is the minimal database connection functionality required. Implemented
+// by *sql.DB.
 type DB interface {
-	Exec(query string, args ...interface{}) error
-	Query(query string, args ...interface{}) (Rows, error)
 	Close() error
+	Exec(query string, args ...interface{}) (sql.Result, error)
+	Ping() error
+	Prepare(query string) (*sql.Stmt, error)
+	Query(query string, args ...interface{}) (*sql.Rows, error)
```
Comment (asim) [inline sql/sql.go]: How do we return *sql.Rows? We can't implement it. Are we using database/sql to do this?

File: sql/sql.go [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+	QueryRow(query string, args ...interface{}) *sql.Row
 }
 
-type Rows interface {
-	Close() error
-	Next() (Row, error)
+type sqlDb interface {
```
Comment (asim) [inline sql/sql.go]: Why is this an unexported interface?

File: sql/sql.go [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
+	Begin() (*sql.Tx, error)
 }
 
-type Row interface {
-	Scan(v interface{}) error
+type sqlTx interface {
```
Comment (asim) [inline sql/sql.go]: Why is this an unexported interface? Why does it exist?

File: sql/sql.go (hunk 1 above)
Comment (nkprince007) [reply sql/sql.go]: Yeah. We should.

File: sql/sql.go (hunk 3 above)
Comment (asim) [reply sql/sql.go]: This is a sql package.... it does not make sense to define internal interfaces like this. It's either a Transaction interface or it doesn't exist.

**Pull request merged**